### PR TITLE
Project Extent | Edit project extent

### PR DESCRIPTION
**Pull request by mddilley**
Closes https://github.com/cityofaustin/atd-data-tech/issues/4906

This PR adds an edit button to the project summary button which opens a [full screen dialog](https://material-ui.com/components/dialogs/#full-screen-dialogs) that shows a larger map for editing. I reused the `NewProjectMap` for the edit map since it just needs setters and getters for state that holds selected layer IDs and the project feature collection.

#### Project Summary with map and edit button
Screen Shot 2021-01-27 at 11 40 56 AM

#### Full screen dialog edit view
Screen Shot 2021-01-27 at 11 41 24 AM

#### Project Summary after save
Screen Shot 2021-01-27 at 11 41 28 AM

### CODE REVIEW

File: moped-editor/src/views/projects/newProjectView/NewProjectMap.js [hunk 1]
```diff
@@ -110,7 +120,7 @@ const NewProjectMap = ({
         {...viewport}
         ref={mapRef}
         width="100%"
-        height={500}
+        height="60vh"
```
Comment (sergiogcx) [inline moped-editor/src/views/projects/newProjectView/NewProjectMap.js:123]: That was literally the first thing I looked up :) 👍

File: moped-editor/src/views/projects/projectView/ProjectSummary.js [hunk 2]
```diff
@@ -110,9 +112,22 @@ const ProjectSummary = () => {
...
+            </>
+          )}
+          {isEditing && (
+            <ProjectSummaryEditMap
+              projectId={projectId}
               selectedLayerIds={project_extent_ids}
               projectExtentGeoJSON={project_extent_geojson}
+              isEditing={isEditing}
+              setIsEditing={setIsEditing}
+              refetchProjectDetails={refetch}
```
Comment (sergiogcx) [inline moped-editor/src/views/projects/projectView/ProjectSummary.js:130]: Just impressive how easy you make it look 👍

File: moped-editor/src/views/projects/projectView/ProjectSummaryEditMap.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      handleClose();
+    });
+  };
+
+  return (
+    <Dialog
+      fullScreen
+      open={isEditing}
+      onClose={handleClose}
+      TransitionComponent={Transition}
```
Comment (sergiogcx) [inline moped-editor/src/views/projects/projectView/ProjectSummaryEditMap.js:75]: TIL thanks to you! :) Transition objects

**Review by sergiogcx — APPROVED** [review]
This is brilliant! It works flawlessly 🗺️ 💯

File: moped-editor/src/views/projects/projectView/ProjectSummary.js (hunk 2 above)
Comment (mateoclarke) [reply moped-editor/src/views/projects/projectView/ProjectSummary.js:130]: ➕

**Review by mateoclarke — APPROVED** [review]
Screen Shot 2021-01-29 at 1 45 15 AM

File: moped-editor/src/views/projects/projectView/ProjectSummaryEditMap.js (hunk 3 above)
Comment (mateoclarke) [reply moped-editor/src/views/projects/projectView/ProjectSummaryEditMap.js:75]: same

**Comment by amenity** [comment]
Would also like @johnclary to peep this one.

**Comment by amenity** [comment]
@mddilley so sorry for the delay getting this approved. It looks great — splendid, even! 🙌 ✏️ 🗺️ ✨

**Review by amenity — APPROVED** [review]

**Pull request merged**